esmvalcore/cmor/table.py

```diff
@@ -34,40 +35,181 @@ def read_cmor_tables(cfg_developer=None):
+    default: object
+        Default table to look variables on if not found
+
+    alias: list[list[str]]
+        List of known aliases for variables
+
+    strict: bool
+        If False, will look for a variable in other tables if it can not be
+        found in the requested one
+
+    """
+
+    def __init__(self, default, alias, strict):
+        if alias is None:
+            alias = ""
+        self.default = default
+        self.alias = alias
+        self.strict = strict
+        self.tables = {}
+
+    def get_table(self, table):
+        """
+        Search and return the table info.
+
+        Parameters
+        ----------
+        table: basestring
+            Table name
+
+        Returns
+        -------
+        TableInfo
+            Return the TableInfo object for the requested table if
+            found, returns None if not
+
+        """
+        return self.tables.get(table)
+
+    def get_variable(self, table_name, short_name, derived=False):
+        """
+        Search and return the variable info.
+
+        Parameters
+        ----------
+        table_name: basestring
+            Table name
+        short_name: basestring
+            Variable's short name
+        derived: bool, optional
+            Variable is derived. Info retrieval is less strict
+
+        Returns
+        -------
+        VariableInfo
+            Return the VariableInfo object for the requested variable if
+            found, returns None if not
+
+        """
+        aliases = self._get_aliases(short_name)
+
+        table = self.get_table(table_name)
+        if table:
+            for alias in aliases:
+                try:
+                    return table[alias]
+                except KeyError:
+                    pass
+
+        var_info = self._look_in_all_tables(aliases)
+        if not var_info:
+            var_info = self._look_in_default(derived, aliases, table_name)
+        if var_info:
+            var_info = self._update_mip_info(table_name, var_info)
+
+        return var_info
+
+    def _look_in_default(self, derived, aliases, table_name):
+        var_info = None
+        if (not self.strict or derived):
+            for alias in aliases:
+                var_info = self.default.get_variable(table_name, alias)
+                if var_info:
+                    break
+        return var_info
+
+    def _look_in_all_tables(self, aliases):
+        var_info = None
+        if not self.strict:
+            for alias in aliases:
+                var_info = self._look_all_tables(alias)
+                if var_info:
+                    break
+        return var_info
+
+    def _get_aliases(self, short_name):
+        aliases = [short_name]
+        for alias_list in self.alias:
+            if short_name in alias_list:
+                aliases.extend(
+                    [alias for alias in alias_list if alias not in aliases])
+        return aliases
+
+    def _update_mip_info(self, table_name, var_info):
```

Comment:
```suggestion
    def _use_frequency_from_mip(self, table_name, var_info):
```
or something similar might describe better what the function does